### PR TITLE
updated build backend

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- updated pep-517 build backend according to https://python-poetry.org/blog/announcing-poetry-1-1-0.html
- change is not strictly necessary, but the old package name is deprecated since poetry moved their pip builder to a separate package. so sooner or later it is necessary

### CODE REVIEW

**Comment by tmbo** [comment]
🛑   this can't be merged yet, build system requires the poetry version which isn't mentioned anywhere after switching the builder...https://github.com/python-poetry/poetry/issues/3469

**Review by joejuzl — APPROVED** [review]

File: scripts/poetry-version.sh [hunk 1]
```diff
@@ -1,13 +1,17 @@
 #!/usr/bin/env python
 
-"""Extract the poetry version from pyproject.toml. Used in e.g. github workflows."""
+"""Extract the poetry version from .github/poetry_version.txt. Used in e.g. github workflows."""
 
+import pathlib
```
Comment (tczekajlo) [inline scripts/poetry-version.sh:5]: Does it require python 3?

Comment (tczekajlo) [reply scripts/poetry-version.sh:5]: This should help

```
  docker:
    name: Build Docker
    runs-on: ubuntu-20.04
```

Comment (joejuzl) [reply scripts/poetry-version.sh:5]: in `continous-integration.yml`?

Comment (tczekajlo) [reply scripts/poetry-version.sh:5]: Yes :)

**Review by tczekajlo — APPROVED** [review]
LGTM 🎖️

**Pull request merged**